lib/dfu.js

```diff
@@ -214,5 +161,45 @@ var that = module.exports = {
 		}
 	},
 
+	_validateSegmentSpecs: function(segmentName) {
```

Comment:
I'm using the term "segment" to refer to the sections of memory for each piece of data we read/write from/to the device (server key, user firmware, etc.). I'm not sure if there's a better name for this, but I thought it worth explaining in case it's not immediately clear.